server/store/sqlstore/channel.go

```diff
@@ -0,0 +1,58 @@
+package sqlstore
+
+import (
+	"strings"
+
+	sq "github.com/Masterminds/squirrel"
+
+	"github.com/mattermost/mattermost/server/public/model"
+
+	"github.com/pkg/errors"
+)
+
+var escapeLikeSearchChar = []string{
+	"%",
+	"_",
+}
+
+func sanitizeSearchTerm(term string) string {
+	const escapeChar = "\\"
+
+	term = strings.ReplaceAll(term, escapeChar, "")
+
+	for _, c := range escapeLikeSearchChar {
+		term = strings.ReplaceAll(term, c, escapeChar+c)
+	}
+
+	return term
+}
+
+func (ss SQLStore) SearchChannelsByPrefix(prefix string) ([]*model.Channel, error) {
```

Comment:
Unsure if we have a better way of doing this. Isn't there a method to search channels in the plugin API?

Reply:
There _is_ a `SearchChannels()` method, but it's limited to one team at a time, and it will match text anywhere within the name, display name, or channel purpose. The implementation here will only do a prefix match, and will do it across all teams.

I could switch to use the plugin api `SearchChannels()`, but the behavior will be different. WDYT, should I switch to use the api version? I think I'd need to call it multiple times to iterate over all teams.